src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -828,4 +840,12 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
       .copy(labels = allLabels)
       .copy(userJupyterExtensionConfig = updatedUserJupyterExtensionConfig)
   }
+
+  private[service] def processClusterImages(clusterRequest: ClusterRequest): Set[ClusterImage] = {
+    // Default to the configured default image if not specified in the request
+    // TODO should we validate the image somehow?
```

Comment:
I wasn't sure what kind of docker validation to do up front before we try to create the cluster. If the user passes a bogus docker image it'll fail in the init script -- is this OK or should we do some kind of validation sooner?

Reply:
what type of validation can we actually do up front? in it's current state, i'm not sure what the behavior would look like if we let it fail in the init script, but i'd hate for us to get user bug reports and be on the hook for debugging what's wrong with custom docker images. as long as we can surface an error message that's useful i think this is ok

Reply:
Yeah agreed - we could parse the docker images but that seems kind of brittle. I'm ok with letting it fail at cluster init time (it should put the cluster in Error state and surface an error message with the init logs).